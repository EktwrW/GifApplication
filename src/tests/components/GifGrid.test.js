import React from 'react';
import { useFetchGifs } from '../../hooks/useFetchGifs';
import '@testing-library/jest-dom';
import { GifGridItem } from '../../components/GifGridItem';

const { GifGrid } = require("../../components/GifGrid");
const { shallow } = require("enzyme");
jest.mock('../../hooks/useFetchGifs');


describe('Pruebas en <GifGrid />', () => {

    const category = 'megadeth';
    
    
    test('debe mostrarse correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        
        const wrapper = shallow( <GifGrid category={ category } /> );

        expect( wrapper ).toMatchSnapshot();
    })

    test('debe mostrar items cuando se cargan imagenes useFetchGifs', () => {
        
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquiercosa/ConstantSourceNode.jpg',
            title: 'Cualquier cosa'
        },
        {
            id: '123',
            url: 'https://localhost/cualquiercosa/ConstantSourceNode.jpg',
            title: 'Cualquier cosa'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow( <GifGrid category={ category } /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );

    })
    
})