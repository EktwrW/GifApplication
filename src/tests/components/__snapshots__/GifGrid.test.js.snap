// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3>
    megadeth
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquiercosa/ConstantSourceNode.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Cualquier cosa"
      url="https://localhost/cualquiercosa/ConstantSourceNode.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifGrid /> debe mostrarse correctamente 1`] = `
<Fragment>
  <h3>
    megadeth
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
